datatree/tests/test_datatree.py

```diff
@@ -63,7 +63,7 @@ def test_empty(self):
         assert dt.name == "root"
         assert dt.parent is None
         assert dt.children == {}
-        xrt.assert_identical(dt.ds, xr.Dataset())
+        xrt.assert_identical(dt.to_dataset(), xr.Dataset())
```

Comment:
I'm not sure whether it's better to use `.ds` or `.to_dataset()` in these comparisons in the tests.

If it weren't for the [explicit type comparisons](https://github.com/pydata/xarray/blob/0479b2d50ace39b02cf58a7131d91b64ad484e90/xarray/testing.py#L108) in `xarray.testing.assert_*` then these asserts would pass without me needing to change `.ds` to `.to_dataset`.